source/whats-new.txt

```diff
@@ -36,43 +36,109 @@ Upcoming Breaking Changes
-  commands are deprecated in MongoDB v6.2 and later. Use the ``$collStats`` aggregation
-  pipeline stage to retrieve the information provided by these methods
-  instead. You can run the aggregation as shown in the following code example:
-
-  .. code-block:: java
-
-     collection.aggregate(Arrays.asList(
-                      new Document("$collStats",
-                          new Document("storageStats", new Document()))));
-
-  To determine whether the collection is a capped collection, access the value
-  of the ``storageStats.capped`` field returned by this example aggregation.
-
-  To learn more about the ``$collStats`` aggregation operator, see the
-  :manual:`$collStats (aggregation) </reference/operator/aggregation/collStats/>`
-  Server manual entry.
-
 - `MapReduceIterable <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MapReduceIterable.html>`__
   and map-reduce methods on `MongoCollection <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#mapReduce(com.mongodb.client.ClientSession,java.lang.String,java.lang.String)>`__
   are deprecated in MongoDB v4.2 and later. They will be replaced by the
   aggregation framework and removed in a future release.
 - Beginning with v5.0, the {+driver-short+} will require Java 11 or later.
+- Beginning with v5.0, several Stream API methods and classes will be
+  removed. To learn which methods will be removed, see :ref:`version-4.11`.
 
 .. _version-4.11:
 
 What's New in 4.11
 ------------------
 
+.. important:: Deprecation Notice
+
+   - The ``getStats()`` and ``isCapped()`` instance methods of the
+     ``DBCollection`` class are deprecated. The corresponding server
+     commands are deprecated in MongoDB v6.2 and later. Use the ``$collStats`` aggregation
+     pipeline stage to retrieve the information provided by these methods
+     instead. You can run the aggregation as shown in the following code example:
+
+     .. code-block:: java
+
+        collection.aggregate(Arrays.asList(
+                         new Document("$collStats",
+                             new Document("storageStats", new Document()))));
+
+     To determine whether the collection is a capped collection, access the value
+     of the ``storageStats.capped`` field returned by this example aggregation.
+
+     To learn more about the ``$collStats`` aggregation operator, see the
+     :manual:`$collStats (aggregation) </reference/operator/aggregation/collStats/>`
+     Server manual entry.
+
+   - The following network address-related methods are deprecated:
+
+     - ``getSocketAddress()`` and ``getSocketAddresses()`` methods from the `ServerAddress <{+api+}/apidocs/mongodb-driver-core/com/mongodb/ServerAddress.html>`__ class
+     - ``getUnixSocketAddress()`` from the `UnixServerAddress <{+api+}/apidocs/mongodb-driver-core/com/mongodb/UnixServerAddress.html>`__ class
+
+     - The following Stream API methods and classes are deprecated and are
```

Comment:
We should elaborate on this section.  The most important part is to show how to configure Netty with replacement API.  I think it's in the ticket but it looks something like this:

```
   clientSettingsBuilder.transportSettings(
        TransportSettings.nettyBuilder()
        ...
        . build())
```

Reply:
I'm not that familiar with how this works. Could author the exact code example you want to include in the docs for us? I think that could help us expedite this task.
Would the explanatory text along the lines of "To configure Netty with a replacement API, pass the following code to your ClientSettings.Builder" work for you?

Reply:
Maybe instead: "To configure Netty with a non-deprecated API"?
 

Deprecated code:

```
import com.mongodb.connection.netty.NettyStreamFactoryFactory;

...

        MongoClientSettings settings = MongoClientSettings.builder()
                .streamFactoryFactory(NettyStreamFactoryFactory.builder().build())
                .build();
```

New code

```
import com.mongodb.connection.TransportSettings;

...

        MongoClientSettings settings = MongoClientSettings.builder()
                .transportSettings(TransportSettings.nettyBuilder().build())
                .build();

```